pkgs/build-support/dotnet/make-nuget-deps/make-source-independent.sh

```diff
@@ -0,0 +1,20 @@
+# shellcheck shell=bash
+
+# NuGet.org edits packages by signing them during upload, which makes
+# those packages nondeterministic. Strip out the signature file.
+# This does ruin our ability to `dotnet nuget verify`, so we'll do
+# that if possible; current advice is not to verify on macOS
+# (see https://github.com/NuGet/Home/issues/11986).
+
+if [ -n "$DOTNET_EXE" ]; then
+    export DOTNET_CLI_HOME
+    DOTNET_CLI_HOME=$(mktemp -d)
+    DOTNET_NOLOGO=1 DOTNET_CLI_TELEMETRY_OPTOUT=1 "$DOTNET_EXE" nuget verify "${ORIGINAL_NUPKG:?}"
+    rm "$DOTNET_CLI_HOME"
+    unset DOTNET_CLI_HOME
+fi
+
+tmp_output=$(mktemp tmp-XXXXXXXX.zip)
+cp "${ORIGINAL_NUPKG:?}" "$tmp_output"
+zip -d "$tmp_output" ".signature.p7s"
```

Comment:
I don't really know anything about the zip format, but I'm a little surprised this works reliably.

So presumably the nuget server simply adds a file to the zip you upload, without ever recompressing it or anything?  I wonder if that's somehow guaranteed?

If that's the case I guess this should be a lot faster than repacking it.

Reply:
I'm really struggling to find where this happens. NuGetGallery itself is net472, even, so my non-Windows IDE is crippled, but the only usage of [`SigningSpecifications`](https://github.com/NuGet/NuGet.Client/blob/128a5066b1438627ac69a2ffe9de564b2c09ee4d/src/NuGet.Core/NuGet.Packaging/Signing/Specifications/SigningSpecificationsV1.cs#L12) in NuGetGallery appears to be in test code. I'll keep digging and I'll ask the NuGet team.

Reply:
I have asked, but I *believe* this ultimately happens at https://github.com/NuGet/NuGet.Client/blob/128a5066b1438627ac69a2ffe9de564b2c09ee4d/src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveIOUtility.cs#L518 , which is fortunately extremely explicit and appears to have the behaviour we want! No guarantees of whether they'll change it in the future, of course.

Reply:
Thanks for digging that up.  Even if it does stop working it shouldn't actually break anything, unless we also move to fixed output derivations.